src/Validation/Validator/ArrayValidator.php

```diff
@@ -56,6 +56,11 @@ protected function doValidation(
 
         $itemsSchema = $this->schema->getItemsSchema();
         foreach ($data as $itemIndex => $itemData) {
+            if (! is_int($itemIndex)) {
+                $resultBuilder->addUnexpectedViolation($breadCrumbPath);
```

Comment:
`UnexpectedViolation` znamena, ze je prvek neocekavany. Pokud to tak bylo mysleno, pak nesedi `breadCrumbPath`, protoze nemiri na konkretni prvek, ktery je neocekavany. To co chces ve skutecnosti  pridat je stejna chyba jako na radku 42, protoze to co jsi dostal ve skutecnosti neni pole a pak udelat break.

Reply:
A ma-li to byt opravdu korektni, tak bychom take meli kontrolovat, ze indexy jsou vzrustajici integery zacinajici nulou.